docs/developer/pull-requests.md

```diff
@@ -0,0 +1,128 @@
+# Pull Request guidelines
+This document outlines some good practices to follow when creating or reviewing PRs. These are all guidelines and not hard rules, so take them as so.
+
+## As an author
+
+### PR size
+
+We don't have a hard limit since that would be too prescriptive and inflexible. However, the recommendation is to not exceed 500 lines of code.
+
+Split them if necessary, following the Single Responsibility Principle.
+
+That said, beware that not all PRs are created equal. It doesn’t produce the same cognitive load to review a PR with 10 new lines of actual code and 500 lines of tests and data files than 200 lines of actual code changes over 20 different files. Prioritize making the review easier, but don’t create extra work for yourself just to align with a random self imposed number. When in doubt, talk to your reviewers and find something that you both feel comfortable with.
```

Comment:
I think we need to be careful here. Its worded as if test code is exempt, entirely, from the 500 LoC recommendation. I don't think that's wise. Test code is just code. It still needs reviewing like any other piece of code. Dependent on how the tests are written, i.e. if someones tried to be clever and de-duplicate a bunch of stuff, it could be low or high cognitive load.

What do we think about making it abundantly clear with the following

> The recommended maximum lines of change is 500. Split PRs as necessary and be guided by the Single Responsibility Principle. Put yourself in the reviewers shoes and consider the cognitive load on the reviewer.
>
> For example, a PR with 1000 LoC touching 20 files could be easy to review if the changes are grammar or could be difficult to review if there are functional changes in loosely related components.